src/rust/engine/process_execution/src/lib.rs

```diff
@@ -57,4 +61,8 @@ pub struct ExecuteProcessResult {
   pub stdout: Bytes,
   pub stderr: Bytes,
   pub exit_code: i32,
+
+  // It's unclear whether this should be a Snapshot or a digest of a Directory. A Snapshot is handy,
```

Comment:
I think that in the case of a remote process, materializing a `Snapshot` with `PathStat` objects will have a non-zero cost (recursively fetching the directory structure, basically).

... which... probably applies to a bunch of usecases where we're currently talking about passing around `Snapshots`. Hm. It might be the case that we need to actually reify "digest" into the python API, so that we avoid using `Snapshots` except when we actually need their listings? Relates to #5502 certainly.

Reply:
Let's continue that discussion there :)